### PR TITLE
Added a constructor to set the balance

### PR DESCRIPTION
**Pull request by Ryan-Gordon**
Also added a test; on the `beforeEach` in TestGameJam.js we set an initialBalance and then verify that value. 

Is this the intended way to take on this task (having one balance for the contract) or should it be a mapping of balances to addresses ?

### CODE REVIEW

File: migrations/2_deploy_contracts.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ var GameJam = artifacts.require("./GameJam.sol");
 module.exports = async (deployer, network) => {
   console.log(`deploying on ${network} network`)
 
-  deployer.deploy(GameJam);
+  await deployer.deploy.apply(deployer, [GameJam, 42])
```
Comment (Ryan-Gordon) [inline migrations/2_deploy_contracts.js]: Is there a better way I can do this ?
I could do something like 

```suggestion
  const initialBalance = 42
  await deployer.deploy.apply(deployer, [GameJam, initialBalance])
```
But is there an even better way ?

File: test/TestGameJam.js [hunk 2]
```diff
@@ -3,18 +3,26 @@ const GameJam = artifacts.require("../contracts/GameJam.sol");
...
   })
 
   it("whitelisted admin votes on a ballot", async () => {
     const competitorAdded = await gameJam.addCompetitor(admin, validIpfsHash)
 
     assert.equal(competitorAdded.logs[0].args.competitor, admin, "Failed to add competitor")
   });
+
+  it("balance variable should match the initialBalance passed in constructor", async () => {
+    const contractBalance = await gameJam.getContractBalance()
```
Comment (Ryan-Gordon) [inline test/TestGameJam.js]: Do I definitely need a getter to test `gameJam`'s contract balance?

File: contracts/GameJam.sol [hunk 3]
```diff
@@ -1,10 +1,18 @@
...
 contract GameJam {
+  uint public balance;
+
   mapping(address => string) public competitors;
 
   event CompetitorAdded(address competitor);
 
+  // Create a GameJam with a _balance
+  // which will be the amount to be distributed to winners
+  constructor(uint _balance) public {
```
Comment (Ryan-Gordon) [inline contracts/GameJam.sol]: Rather than explicitly having a `_balance` in the constructor, should I be making use of the `payable` keyword?

Comment (PaddyMc) [reply contracts/GameJam.sol]: Na this is perfect, the `payable` keyword we use for passing eth to a function.

The idea is we have a management contract that instantiates this contract. And on the instantiation function, we will have the `payable` keyword

File: migrations/2_deploy_contracts.js (hunk 1 above)
Comment (PaddyMc) [reply migrations/2_deploy_contracts.js]: Please do it this way, we want to avoid `magic numbers` :)

File: test/TestGameJam.js (hunk 2 above)
Comment (PaddyMc) [reply test/TestGameJam.js]: When we declare public variables default getters are generated.

So we can:
 
```javascript
const contractBalance = await gameJam.balance();
```

And completely remove the `getContractBalance()`

File: contracts/GameJam.sol [hunk 4]
```diff
@@ -15,4 +23,9 @@ contract GameJam {
 
     emit CompetitorAdded(competitor);
   }
+
+  // Getter for the balance variable
+  function getContractBalance() public view returns(uint) {
```
Comment (PaddyMc) [inline contracts/GameJam.sol]: We can remove this completely  please see comment below <3

File: migrations/2_deploy_contracts.js (hunk 1 above)
Comment (PaddyMc) [reply migrations/2_deploy_contracts.js]: Not sure if there's a better way than this, maybe @CaelRowley can learn us 😄

File: contracts/GameJam.sol (hunk 3 above)
Comment (PaddyMc) [reply contracts/GameJam.sol]: Actually after investigating we need the `payable` function in the constructor.

like this:

```solidity
constructor (uint _balance)  payable public {
```

Then we can get the balance like this:

```solidity
address(this).balance
```

✊

File: migrations/2_deploy_contracts.js (hunk 1 above)
Comment (Ryan-Gordon) [reply migrations/2_deploy_contracts.js]: Changed this to the suggested change and now setting it to zero initially. We may need to change the setting to zero for testing purposes, what do you think?

File: test/TestGameJam.js (hunk 2 above)
Comment (Ryan-Gordon) [reply test/TestGameJam.js]: Removed getContractBalance
Now using `balance()` in the test

File: contracts/GameJam.sol (hunk 3 above)
Comment (Ryan-Gordon) [reply contracts/GameJam.sol]: Added payable to constructor and now have a .sol test ✊

File: test/TestGameJam.js (hunk 2 above)
Comment (Ryan-Gordon) [reply test/TestGameJam.js]: Added 
`const contractBalance = await gameJam.balance();`
and removed the `getContractBalance`

File: migrations/2_deploy_contracts.js (hunk 1 above)
Comment (PaddyMc) [reply migrations/2_deploy_contracts.js]: Tbh it's perfect for the moment! gg

**Pull request merged**